tasks/milmove.py

```diff
@@ -27,11 +35,306 @@ def on_start(self):
             self.interrupt()
 
     @task
-    def get_user_info(self):
+    def stop(self) -> None:
         """
-        Gets the user info for the currently logged in user.
+        This ensures that at some point, the user will stop running the tasks in this task set.
+        """
+        self.interrupt()
+
+    @tag("getUserInfo")
+    @task(2)
+    def get_user_info(self) -> Optional[JSONObject]:
+        """
+        Gets the user info for the currently logged-in user.
+        :return: logged-in user, or None if there is an error
         """
         url, request_kwargs = self.request_preparer.prep_internal_request(endpoint="/users/logged_in")
 
         with self.rest(method="GET", url=url, **request_kwargs) as resp:
-            logger.info(f"ℹ️ User email: {resp.js.get('email', 'None')}")
+            resp: RestResponseContextManager
+
+            log_response_info(response=resp)
+
+            if resp.status_code == HTTPStatus.OK:
+                user: JSONObject = resp.js
+
+                logger.info(f"ℹ️ User email: {user['email']}")
+
+                return user
+            else:
+                resp.failure("Unable to get logged-in user.")
+
+                log_response_failure(response=resp)
+
+                return
+
+    @tag("onboardCustomerWorkflow")
+    @task
+    def onboard_customer(self) -> None:
+        """
+        Goes through the basic on-boarding flow for a customer.
+
+        We're going to be re-using the same endpoint for several requests in this workflow.
+        Because of that, we won't use the api fake data generator because we would end up having
+        to delete a lot of the data it gave back in order to simulate the step-by-step process
+        that on-boarding actually takes. Instead, we'll use the lower-level `MilMoveData` which is
+        what the api fake data generate uses internally because it gives us more control. It's
+        essentially a wrapper around faker, but we'll still use it to remain more consistent in
+        the faker helper functions we use for different data types.
+
+        Another thing to note is that because we'll be hitting a lot of endpoints in this load test,
+        we'll pass task_name values to the logging functions whenever we hit an endpoint to make it
+        easier to see in the logs what is happening.
+        """
+        user = self.get_user_info()
+
+        if user is None:
+            return  # get_user_info will log the error
+
+        service_member_id = user["service_member"]["id"]
+
+        milmove_faker = MilMoveData()
+
+        # First let's create the profile.
+        payload = {
+            "id": service_member_id,
+            "affiliation": "ARMY",
+            "edipi": "1234567890",
+            "rank": "E_1",
+        }
```

Comment:
A few are based on ENUMs and I noticed the code for reading the enums and picking a random value was at the parser level so I didn't want to try to piece that together. I do think if we implemented the thing you mentioned about passing in what params you want, it would be better.